harness/determined/pytorch/experimental/_torch_batch_process.py

```diff
@@ -24,12 +24,21 @@
 class TorchBatchProcessorContext(pytorch._PyTorchReducerContext):
     def __init__(self, core_context: core.Context, storage_path: str) -> None:
         super().__init__()
-        self._core_context = core_context
-        self.distributed = self._core_context.distributed
+        self.core_context = core_context
+        self.distributed = self.core_context.distributed
         self.device = get_default_device(core_context)
         self._tensorboard_path = core_context.train.get_tensorboard_path()
         self._storage_path = storage_path
         self._use_default_storage = False
+        self._hparams = None  # type: Optional[Dict[str, Any]]
+
+    def get_hparams(self) -> Dict[str, Any]:
+        if self._hparams is None:
+            info = det.get_cluster_info()
+            assert info, "Must run TorchBatchProcessor in a cluster to run get_hparams()"
+            self._hparams = info.trial.hparams
```

Comment:
you should also be testing that `info.task_type == "TRIAL"` before reading `info.trial`

Reply:
added